model/src/main/proto/exploration.proto

```diff
@@ -420,3 +420,32 @@ enum CheckpointState {
   // Progress made in the exploration is not saved.
   CHECKPOINT_UNSAVED = 3;
 }
+
+// Corresponds to a item selection answer that user has selected.
+message ItemSelectionAnswerState {
+  repeated int32 selected_indexes = 1;
+}
+
+// Corresponds to a raw representation of the current answer entered by the user
+// which is used to retain state on configuration changes
+message UserAnswerState {
+  AnswerErrorCategory answer_error_category = 1;
+  oneof answer_input_type{
+    ItemSelectionAnswerState item_selection = 2;
+
+    // A raw answer entered by user in a text-based interactions.
+    string textual_answer = 3;
+  }
+}
+
+// Represents categories of errors that can be inferred from a pending answer.
+enum AnswerErrorCategory {
+  // Corresponds to the pending answer having no error.
+  NO_ERROR_UNSPECIFIED = 0;
+
+  // Corresponds to errors that may be found while the user is trying to input an answer.
+  REAL_TIME = 1;
+
+  // Corresponds to errors that may be found only when a user tries to submit an answer.
+  SUBMIT_TIME = 2;
+}
```

Comment:
```suggestion
enum AnswerErrorCategory {
   // The error is unknown or not specified.
 ERROR_UNSPECIFIED = 0;
  
  // Corresponds to the pending answer having no error.
  NO_ERROR = 1;

  // Corresponds to errors that may be found while the user is trying to input an answer.
  REAL_TIME = 2;

  // Corresponds to errors that may be found only when a user tries to submit an answer.
  SUBMIT_TIME = 3;
}
```
no_error, and the unspecified state do not mean the same thing in protobuf. unspecified is the default fallback for enums, and unrecognized enum is created by the compiler for a value that is out of bounds(-1), or unknown.

Reply:
Thanks.